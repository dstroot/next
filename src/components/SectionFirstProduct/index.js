import React from 'react';

//compnent
import Card from '../Card';

const SectionFirstProduct = () => {
  return (
    <div className="nxt_bg-gray py-6">
      <div className="container text-center">
        <div className="row">
          <div className="col-md-12">
            <h1 className="nxt_heading-xlarge">
              Introducing our first product
            </h1>
            <Card />
          </div>
        </div>
      </div>
    </div>
  );
};

export default SectionFirstProduct;
