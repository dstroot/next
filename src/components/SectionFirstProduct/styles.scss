// custom css
.slab {
  font-family: 'Zilla Slab', serif;
  font-size: calc(40px + 1.5vw + 0.5vmin);
}

.circular {
  font-family: CircularStd;
  font-size: 24px;
}
