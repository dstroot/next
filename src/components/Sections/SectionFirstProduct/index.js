import React from 'react';

//compnent
import Card from '../../Card';

// images
import Tree from '../../Card/media/evergreen.svg';

const SectionFirstProduct = () => {
  return (
    <div className="nxt_bg-gray py-6">
      <div className="container text-center">
        <div className="row">
          <div className="col-md-12">
            <h1 className="nxt_heading-xlarge">
              Introducing our first product
            </h1>
            <Card
              img={Tree}
              title="Next Deferred Income Annuity"
              content="Create a future income stream you can count on for the rest of your life by saving over time."
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default SectionFirstProduct;
