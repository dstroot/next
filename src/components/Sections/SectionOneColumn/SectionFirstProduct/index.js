import React from 'react';

// animation
import { TweenMax, TimelineMax } from 'gsap';
import ScrollMagic from 'scrollmagic';
import 'scrollmagic/scrollmagic/uncompressed/plugins/animation.gsap';
import 'scrollmagic/scrollmagic/uncompressed/plugins/debug.addIndicators';

//compnent
import Card from '../../../Card';
import SectionOneColumn from '..';
import Button from '../../../Buttons/Button';

// images
import Tree from '../../../Card/media/evergreen.svg';
import Mountains from '../../../Card/media/mountains.svg';

class SectionFirstProduct extends React.Component {
  constructor(props) {
    super(props);

    // GSAP
    this.cardTl = new TimelineMax();
    this.cards = [];

    // ScrollMagic
    this.controller = new ScrollMagic.Controller();
  }

  componentDidMount() {
    // GSAP Timeline
    this.cardTl.staggerFrom(
      this.cards,
      0.5,
      {
        scale: 0,
      },
      0.1
    );
    // ScrollMagic
    new ScrollMagic.Scene({
      triggerElement: '#cardScrollStarts',
      // triggerHook: 0.65,
    })
      .setTween(this.cardTl)
      .addIndicators() // for debugging
      .addTo(this.controller);
  }

  render() {
    return (
      <SectionOneColumn
        styleBackground="bg-light"
        heading="Introducing our first product"
        noFadeContent={
          // MAYBE PASS IN CARD-DECK DIV CONTAINING ALL CARDS, THEREFOR WE CAN CONTROL STAGGER ANIMATION AND STYLING?
          <div class="card-deck">
            <Card
              img={Mountains}
              title="Next Term Life"
              content={
                <>
                  <p>
                    Help keep your loved ones protected with coverage that
                    adapts with your lifeâ€™s biggest moments.
                  </p>
                  <br />
                  <p>
                    <strong>Coming Soon! </strong>
                  </p>
                </>
              }
            />

            <Card
              img={Tree}
              title="Next Deferred Income Annuity"
              content="Create a future income stream you can count on for the rest of your life by saving over time."
              button={
                <Button
                  path="/nextdeferredincomeannuity"
                  buttonText="Learn More"
                />
              }
            />
          </div>
        }
      />
    );
  }
}

export default SectionFirstProduct;
