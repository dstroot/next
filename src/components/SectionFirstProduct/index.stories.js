import React from 'react';
import { storiesOf } from '@storybook/react';
import SectionFirstProduct from '../SectionFirstProduct';

storiesOf('NEXT|sections|SectionFirstProduct', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!
      
      ~~~js
      console.log("hello");
      ~~~
      `,
    },
  })
  .add('show SectionFirstProduct', () => <SectionFirstProduct />);
