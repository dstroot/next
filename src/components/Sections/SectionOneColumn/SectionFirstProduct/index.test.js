import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import SectionFirstProduct from '../SectionFirstProduct';

describe('SectionFirstProduct', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<SectionFirstProduct />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<SectionFirstProduct />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
