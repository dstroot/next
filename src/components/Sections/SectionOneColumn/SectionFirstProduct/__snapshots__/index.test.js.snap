// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionFirstProduct it should render 1`] = `
<div
  className="py-6 bg-light"
>
  <div
    className="container text-center"
  >
    <h1
      className="react-reveal  nxt_heading-xlarge "
      style={
        Object {
          "opacity": undefined,
        }
      }
    >
      Introducing our first product
    </h1>
    <h1
      className="react-reveal nxt_heading-large mb-5"
      style={
        Object {
          "opacity": undefined,
        }
      }
    />
    <div
      className="react-reveal"
      style={
        Object {
          "opacity": undefined,
        }
      }
    >
      <div
        className="card mx-auto my-5 card border-dark mb-3 nxt_rounded-corner p-3 nxt_card-width"
      >
        <div
          className="card-body text-center"
        >
          <img
            alt="Evergreen"
            className="mb-2"
            src="evergreen.svg"
          />
          <h5
            className="card-title nxt_heading-small mb-4"
          >
            Next Deferred Income Annuity
          </h5>
          <p
            className="card-text circular mb-5"
          >
            Create a future income stream you can count on for the rest of your life by saving over time.
          </p>
          <a
            href="/nextdeferredincomeannuity"
            onClick={[Function]}
          >
            <button
              className="btn btn-primary rounded-pill nxt_body-small px-4 py-2"
            >
              Learn More
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
